common/predictive-text/worker/correction/context-tracker.ts

```diff
@@ -14,16 +14,93 @@ namespace correction {
   }
 
   export class TrackedContextState {
-    context: Context;
+    // Stores the source Context (as a debugging reference).  Not currently utilized.
+    taggedContext: Context;
+
     tokens: TrackedContextToken[];
-    poppedHead: boolean;
-    pushedTail: boolean;
-    
+    indexOffset: number;
```

Comment:
When is this property read or utilized?

Reply:
It's not utilized for much yet, but it may be important if and when we start supporting phrase-level corrections/predictions.

Worst-case, it does provide useful debugging feedback.

Reply:
In whatever case, we already needed to compute the value.  As it's a simple primitive, it's also quite low-cost to keep around - especially for something that may prove useful for debugging and for extra functionality down the road.

Reply:
This is an excellent write-up! Could you please add a JSDoc for this property, something like this::

```suggestion
    /**
     * How many tokens were removed from the start of the best-matching ancestor.
     * Useful for restoring older states, e.g., when the user moves the caret backwards, we can recover the context at that position.
     */
    indexOffset: number;
```